include/seqan/index/find2_index_approx.h

```diff
@@ -0,0 +1,585 @@
+    {
+        uint8_t const blockIndex2 = std::min(blockIndex + 1, static_cast<uint8_t>(s.u.size()) - 1);
+        bool const goToRight2 = s.pi[blockIndex2] > s.pi[blockIndex2 - 1];
+
+        if (goToRight2)
+            _schemeSearch(delegate, iter, needle, needleLeftIt, needleRightIt, errors, s, blockIndex2, Rev(),
+                          EditDistance());
+        else
+            _schemeSearch(delegate, iter, needle, needleLeftIt, needleRightIt, errors, s, blockIndex2, Fwd(),
+                          EditDistance());
+    }
+
+    if (maxErrorsLeftInBlock > 0)
+    {
+        if (goDown(iter, TDir()))
+        {
+            do
+            {
+                _schemeSearchDeletion(delegate, iter, needle, needleLeftIt, needleRightIt, errors + 1, s, blockIndex,
+                                      TDir());
+            } while (goRight(iter, TDir()));
+        }
+    }
+}
+
+template <typename TDelegate, typename TText, typename TIndex, typename TIndexSpec, typename TText2, size_t nbrBlocks,
+          typename TDir, typename TDistanceTag>
+inline void _schemeSearchChildren(TDelegate & delegate,
+                                  Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > iter,
+                                  TText2 const & needle,
+                                  uint32_t const needleLeftIt,
+                                  uint32_t const needleRightIt,
+                                  uint8_t const errors,
+                                  Search<nbrBlocks> const & s,
+                                  uint8_t const blockIndex,
+                                  uint8_t const minErrorsLeftInBlock,
+                                  TDir const & /**/,
+                                  TDistanceTag const & /**/)
+{
+    bool goToRight = std::is_same<TDir, Rev>::value;
+    if (goDown(iter, TDir()))
+    {
+        uint32_t charsLeft = s.blocklength[blockIndex] - (needleRightIt - needleLeftIt - 1);
+        do
+        {
+            bool delta = !ordEqual(parentEdgeLabel(iter, TDir()),
+                                   needle[goToRight ? needleRightIt - 1 : needleLeftIt - 1]);
+
+            // NOTE (cpockrandt): this might not be optimal yet! we have more edges than in the theoretical model,
+            // since we go down an edge before we check whether it can even work out!
+            if (!std::is_same<TDistanceTag, EditDistance>::value && minErrorsLeftInBlock > 0 &&
+                charsLeft - 1 < minErrorsLeftInBlock - delta)
+                continue;
+
+            int32_t needleLeftIt2 = needleLeftIt - !goToRight;
+            uint32_t needleRightIt2 = needleRightIt + goToRight;
+
+            if (needleRightIt - needleLeftIt == s.blocklength[blockIndex])
+            {
+                // leave the possibility for one or multiple deletions! therefore, don't change direction, etc!
+                if (std::is_same<TDistanceTag, EditDistance>::value)
```

Comment:
does this decision happen often, could it profit from IF_CONSTEXPR, do we have this macro now?

Reply:
@h-2 Can you point to the macro? I can't find it ...

Reply:
I will make a PR for this in the evening. 😏

Reply:
@rrahn Thanks! 😍